weave-js/src/components/Form/TextField.tsx

```diff
@@ -78,34 +79,37 @@ export const TextField = ({
     <Tailwind style={{width: '100%'}}>
       <div
         className={classNames(
-          'relative',
-          textFieldSize === 'medium' ? 'h-32' : 'h-40'
+          'relative rounded-sm',
+          textFieldSize === 'medium' ? 'h-32' : 'h-40',
+          'outline outline-1 outline-moon-250',
+          {
+            'hover:outline-2 [&:hover:not(:focus-within)]:outline-[#83E4EB]':
+              !errorState,
+            'focus-within:outline-2 focus-within:outline-teal-400': !errorState,
+            'outline-2 outline-red-450': errorState,
+            'pointer-events-none opacity-50': disabled,
+          }
         )}>
-        <div
-          className={classNames(
-            'absolute bottom-0 top-0 flex w-full items-center rounded-sm',
-            'outline outline-1 outline-moon-250',
-            disabled
-              ? 'opacity-50'
-              : 'hover:outline hover:outline-2 hover:outline-teal-500/40 focus:outline-2',
-            errorState
-              ? 'outline outline-2 outline-red-450 hover:outline-red-450 focus:outline-red-450'
-              : 'outline outline-1 outline-moon-250 hover:outline-teal-500/40 focus:outline-teal-500/40'
-          )}>
-          {prefix && <div className="text-gray-800 pl-8 pr-1">{prefix}</div>}
+        <div className="absolute bottom-0 top-0 flex w-full items-center rounded-sm">
+          {prefix && (
+            <div
+              className={classNames(
+                'text-gray-800 pr-1',
+                icon ? leftPaddingForIcon : 'pl-8'
```

Comment:
this is what fixes prefix/icon clashing. just needed to make sure we included extra padding for the icon